### PR TITLE
rewards on account history are fixed

### PR DESCRIPTION
**Pull request by MashaYakovenko**

### CODE REVIEW

File: utilities/operations/account/getHistory.js [hunk 1]
```diff
@@ -86,6 +39,78 @@ const getAccountHistory = async (params) => {
...
+            ],
+    },
+  };
+};
+
+const getFilteredApiData = async ({
+  params, count = 0, filteredApiData = [], timestampEnd = 0, skip = 0,
+}) => {
+  let limit = params.limit + 100;
+  if (params.excludeSymbols) limit = TOKEN_WAIV.MAX_LIMIT;
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: if you not overwrite limit you can use ternarry assigment to const

File: utilities/operations/account/getHistory.js [hunk 2]
```diff
@@ -86,6 +39,78 @@ const getAccountHistory = async (params) => {
...
+
+  const apiResponse = await accountHistory(constructApiQuery({
+    params, limit, count, timestampEnd, skip,
+  }));
+  if (apiResponse instanceof Error) return { errorApiResponse: apiResponse };
+
+  filteredApiData.push(..._.filter(
+    apiResponse.data,
+    (el) => !_.includes(params.excludeSymbols, el.symbol),
+  ));
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: if you don't have params.excludeSymbols it's unnecessary filter

File: utilities/operations/account/getHistory.js [hunk 3]
```diff
@@ -86,6 +39,78 @@ const getAccountHistory = async (params) => {
...
+  const timestampEndForQuery = filteredApiData[filteredApiData.length - 1].timestamp + 1;
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: 1) why + 1? you can have multiple records with same timestamp
2) why filteredApiData not apiResponse?

File: utilities/operations/account/getHistory.js [hunk 4]
```diff
@@ -86,6 +39,78 @@ const getAccountHistory = async (params) => {
...
+
+  if (limit >= TOKEN_WAIV.MAX_LIMIT && filteredApiData.length < limit
+    && timestampEnd !== timestampEndForQuery) {
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: 1) split lines by && or || 
2) if more than one logical operator  set condition to const
2) timestampEnd !== timestampEndForQuery remove
3) suggesting:
if(apiResponse.length < limit || filteredApiData.length >= limit) return { filteredApiData };

File: utilities/operations/account/getHistory.js [hunk 5]
```diff
@@ -86,6 +39,71 @@ const getAccountHistory = async (params) => {
...
+}) => {
+  const limit = params.excludeSymbols ? TOKEN_WAIV.MAX_LIMIT : params.limit + 100;
+
+  const apiResponse = await accountHistory(constructApiQuery({
+    params, limit, timestampEnd,
+  }));
+  if (apiResponse instanceof Error) return { errorApiResponse: apiResponse };
+
+  apiResponseData.push(...apiResponse.data);
+  if (apiResponse.data.length < limit || apiResponseData.length === limit) {
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: apiResponseData.length >= limit

File: utilities/operations/account/getHistory.js [hunk 6]
```diff
@@ -86,6 +39,71 @@ const getAccountHistory = async (params) => {
...
+  params, apiResponseData = [], timestampEnd = 0,
+}) => {
+  const limit = params.excludeSymbols ? TOKEN_WAIV.MAX_LIMIT : params.limit + 100;
+
+  const apiResponse = await accountHistory(constructApiQuery({
+    params, limit, timestampEnd,
+  }));
+  if (apiResponse instanceof Error) return { errorApiResponse: apiResponse };
+
+  apiResponseData.push(...apiResponse.data);
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: apiResponseData.push(params.excludeSymbols ? : ..._.filter():  ...apiResponse.data);

File: utilities/operations/account/getHistory.js [hunk 7]
```diff
@@ -86,6 +39,71 @@ const getAccountHistory = async (params) => {
...
+  if (apiResponse.data.length < limit || apiResponseData.length === limit) {
+    return {
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: return {apiResponseData}

**Review by ghost — APPROVED** [review]

**Pull request merged**